templates/CRM/Admin/Form/Preferences/Contribute.tpl

```diff
@@ -24,3 +24,37 @@
  +--------------------------------------------------------------------+
 *}
 {include file="CRM/Form/basicForm.tpl"}
+{literal}
+  <script type="text/javascript">
+    CRM.$(function($) {
+      showHideElement('deferred_revenue_enabled', 'default_invoice_page');
+      $("#deferred_revenue_enabled").click(function() {
+        showHideElement('deferred_revenue_enabled', 'default_invoice_page');
+      });
+      showHideElement('financial_account_bal_enable', 'fiscalYearStart');
+      $("#financial_account_bal_enable").click(function() {
+        showHideElement('financial_account_bal_enable', 'fiscalYearStart');
+      });
+      function showHideElement(checkEle, toHide) {
+        if ($('#' + checkEle).prop('checked')) {
+          $("tr.crm-preferences-form-block-" + toHide).show();
+        }
+        else {
+          $("tr.crm-preferences-form-block-" + toHide).hide();
+        }
+      }
+      $('input[name=_qf_Contribute_next]').on('click', checkPeriod);
+      function checkPeriod() {
+        var speriod = $('#prior_financial_period').val();
+      	var hperiod = '{/literal}{$priorFinancialPeriod}{literal}';
+      	if (((hperiod && speriod == '') || (hperiod && speriod != '')) && (speriod != hperiod)) {
+          if (confirm('Changing the Prior Financial Period may result in problems calculating closing account balances accurately and / or exporting of financial transactions. Do you want to proceed?')) {
+	    return true;
+          } else {
+            return false;
+          }
```

Comment:
`confirm` already returns a boolean, so wrapping it in `if (confirm()) {return true} else {return false}` is just a verbose way of saying `return confirm()`